react/getting_started_with_react/rendering_techniques.md

```diff
@@ -0,0 +1,299 @@
+      <ul>
+        <li>Lion</li>
+        <li>Cow</li>
+        <li>Snake</li>
+        <li>Lizard</li>
+      </ul>
+    </div>
+  );
+}
+~~~
+
+It is perfectly acceptable, but what if we want to render more than just four? It can be tedious and long, and most of the time, we will be dealing with a data structure (like a list) rather than hard coding each animal. You have previously learned that we can embed expressions inside JSX with curly braces. So let us do just that:
+
+~~~javascript
+function App() {
+  const animals = ["Lion", "Cow", "Snake", "Lizard"];
+  return (
+    <div>
+      <h1>Animals: </h1>
+      <ul>
+        {animals.map((animal) => {
+          return <li key={animal}>{animal}</li>;
+        })}
+      </ul>
+    </div>
+  );
+}
+~~~
+
+We have defined an array called `animals` which contains what we have previously written. Now inside our JSX, we used `map` to map over each animal in the list and return the `li` element, adding `animal` as its text. It should now render the same as the previous snippet we wrote. It is now more dynamic and easy to handle.
+
+You may be curious as to what the `key` is in our `<li>` element. We will dive into how keys work in the next lesson. But, to explain briefly, it is to let React know the identity of each element in the list, React must know this information if you are dealing with a dynamic list where you add or remove elements. Since we are only dealing with a static list, it does not matter for now.
+
+### Rendering a List of Components in JSX
+
+<div class="lesson-note" markdown="1">
+We will use `props` here, and you will learn more about them in a future lesson. We will just be writing a simple implementation.
+</div>
+
+~~~javascript
+function ListItem(props) {
+  return <li>{props.animal}</li>
+}
+
+function List(props) {
+  return (
+    <ul>
+      {props.animals.map((animal) => {
+        return <ListItem key={animal} animal={animal} />;
+      })}
+    </ul>
+  );
+}
+
+function App() {
+  const animals = ["Lion", "Cow", "Snake", "Lizard"];
+
+  return (
+    <div>
+      <h1>Animals: </h1>
+      <List animals={animals} />
+    </div>
+  );
+}
+~~~
+
+We have moved our `<ul>` element to a different component called `<List />`. It still returns the `<ul>` element, but we can do a lot more with it as a component.
+
+This component accepts a `props` which is an object containing the `animals` that we defined as a property when we wrote `<List animals={animals}>`. Do note that you can name it anything, for example `<List animalList={animals} />` you will still need to pass the animals to the property, but now you will use `props.animalList` instead of `props.animals`.
+
+We have also created a different component for the `<li>` element called `<ListItem />`, which also accepts a `props`, and using `props.animal` to render the text. This kind of pattern is very useful because a component can have their own state, unlike a regular element. It should now render the same thing. 
```

Comment:
I'm unsure about this sentence. State hasn't been formally introduced yet, and we aren't even using state in any of our components in the code example. 

> This kind of pattern is very useful because a component can have their own state, unlike a regular element.

Reply:
I was thinking of giving a small overview for the next section. Maybe it was a bit lacking in explanation and going over what's intended, but I can see now how this sentence and comparison would be meaningless. Any suggestions on this one?

Reply:
I think we can remove it.

Reply:
Done!